plugins/search-common/src/types.ts

```diff
@@ -31,26 +31,45 @@ export interface SearchQuery {
 /**
  * @beta
  */
-export interface SearchResult {
+export interface Result<TDocument> {
```

Comment:
Is it worth specifying `TDocument extends SearchDocument` here, at this point?

Reply:
Good idea! Done in 21832efef0.